automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala

```diff
@@ -143,6 +143,9 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
       case e: Exception => throw e
     }
 
+    // confirm that we can still list the Deleting cluster
+    Leonardo.cluster.listIncludingDeleted().find(_.clusterName == clusterName).map(_.status) shouldBe Some(ClusterStatus.Deleting)
```

Comment:
Hmmmm.  This is "flaky" in practice.  Sometimes it deletes too quickly.  Any ideas?

Reply:
Ugh. This is a tricky one. We want to "catch" a cluster in Deleting state, but we can't control how fast it deletes in Real Google. This seems like it would be ideally testable with a Mock Google and a Real Sam. But of course we can't do that.

I don't know, if this is flaky then we should probably take it out of swat and settle for unit test coverage (is `notifyClusterDeleted` called at the right time). Unless anyone else has clever ideas...

Reply:
I agree that this can (should, even) be tested in unit instead. In general we should prefer to test in unit over integration where we can, and this particular check (that we can see Deleting clusters) is definitely unit testable.

Reply:
Hm - not sure if moving this to unit test will make it any less flaky.  But I'll try it.

Reply:
You can call delete on LeoService and it'll do the Leo-side work. Then you test it is in deleting, and that notifyDeleted has NOT yet been called. Then you send ClusterDeleted to the monitor for part two. You might have to look up the monitor with system.actorSelection and there may be other mockery required.